### PR TITLE
Move `SelfAttentionEstimator` to `gluonts.nursery`

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* See eg #2466

*Description of changes:* `SelfAttentionEstimator` does not work on basic examples, is essentially untested, and resulted in a few bug reports. For example, this fails

```python
from gluonts.dataset.repository.datasets import get_dataset, dataset_names
from gluonts.mx.model.san import SelfAttentionEstimator
from gluonts.mx.trainer import Trainer

dataset = get_dataset("electricity")

estimator = SelfAttentionEstimator(
    prediction_length=dataset.metadata.prediction_length,
    freq=dataset.metadata.freq,
    trainer=Trainer(epochs=1),
)

predictor = estimator.train(dataset.train)
```

Hence, this PR moves it to the nursery.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Review by jaheba — DISMISSED** [review]

**Comment by jaheba** [comment]
There were no tests?

**Comment by jaheba** [comment]
We might want to adjust https://ts.gluon.ai/stable/getting_started/models.html as well

**Review by jaheba — APPROVED** [review]

**Comment by melopeo** [comment]
@jaheba That's right. There were no tests for this model.
